opslevel/datasource_opslevel_domain.go

```diff
@@ -87,17 +124,10 @@ func (d *DomainDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read example, got error: %s", err))
 		return
 	}
-
-	data.Id = types.StringValue(string(domain.Id))
-	domainAliases, aliasesDiag := types.ListValueFrom(ctx, types.StringType, domain.Aliases)
-	resp.Diagnostics.Append(aliasesDiag...)
-
-	data.Aliases = domainAliases
-	data.Name = types.StringValue(string(domain.Name))
-	data.Description = types.StringValue(string(domain.Description))
-	data.Owner = types.StringValue(string(domain.Owner.Id()))
-	tflog.Trace(ctx, "read an OpsLevel Domain data source")
+	domainDataModel, diags := newDomainDataSourceModelWithIdentifier(ctx, *domain, data.Identifier)
```

Comment:
Ahh so I think we had a slightly different communication of this idea.  I was thinking we'd do something like

```
model, diags := newDomainDataSourceModel(ctx, *domain)
resp.Diagnostics.Append(diags...)
resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("identifier"), data.Identifier)...)
resp.Diagnostics.Append(resp.State.Set(ctx, &domainDataModel)...)
```

This way we only need `NewDomainDataSourceModel` and then actually it could become `NewDomainModel` because it can be reused with resource

Reply:
Ahh i see then you lose `data.Identifier` - so to get that back you do

```
var identifier string
resp.Diagnostics.Append(req.Config.GetAttribute(ctx, path.Root("identifier"), &identifier)...)
resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
```
then you can use `identifier` without it having to actually be in the model